File: Selenium/Project/Activity5.java
import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;

import java.time.Duration;


public class Activity5 {
    WebDriver driver;

    @BeforeClass
    public void beforeMethod() {
        // Set up Firefox driver
        WebDriverManager.firefoxdriver().setup();
        // Create a new instance of the Firefox driver
        driver = new FirefoxDriver();
        // Create the Wait object
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
        // Create the Actions object
        Actions builder = new Actions(driver);

        // Open the page
        driver.get("https://alchemy.hguy.co/jobs/");
        // Print the title of the page

    }

    @Test
    public void Activity5() {
        driver.findElement(By.linkText("Jobs")).click();
        String secondPageTitle = driver.getTitle();
        System.out.println("Job Page Title:" + secondPageTitle);
        if (secondPageTitle.contains("Jobs – Alchemy Jobs")) {
            System.out.println("Job page title Matched");
        } else {
            System.out.println(" Failed");
        }
    }

    @AfterClass
    public void tearDown() {
        driver.quit();
    }
}

File: Selenium/Project/Activity5.py
import pytest
from selenium import webdriver
from selenium.webdriver.common.by import By


# Create a fixture to initialize our webdriver
@pytest.fixture(scope="session")
def setup(request):
    # Initialize the webdriver
    driver = webdriver.Firefox()

    # Get the underlying collection
    session = request.node

    # Pass through the driver to the test class
    for item in session.items:
        cls = item.getparent(pytest.Class)
        setattr(cls.obj, "driver", driver)

    # Execute the test method
    yield

    # Close the browser once the test has ended
    request.addfinalizer(driver.close)


@pytest.mark.usefixtures("setup")
class TestSite:

    def test_navigation(self):
        self.driver.get("https://alchemy.hguy.co/jobs")

        Jobslink = self.driver.find_element(By.LINK_TEXT, "Jobs")
        Jobslink.click()
        #Jobsection = self.driver.find_element(By.CLASS_NAME, "entry-title")
        #print("Heading: ", Jobsection.text)
        GetTitle = self.driver.title
        Expected_title = "Jobs – Alchemy Jobs"

        assert GetTitle == Expected_title, "Title mismatch"

